kernel/src/hil/time.rs

```diff
@@ -1,20 +1,32 @@
 //! Hardware agnostic interfaces for counter-like resources.
 
+use crate::ReturnCode;
+
 pub trait Time {
     type Frequency: Frequency;
 
-    /// Disable any outstanding alarm or timer
-    fn disable(&self);
+    /// Returns the current time in hardware clock units.
+    fn now(&self) -> u32;
 
-    /// Returns whether a timer or alarm is currently armed
-    fn is_armed(&self) -> bool;
+    /// Returns the wrap-around value of the clock.
+    ///
+    /// The maximum value of the clock, at which `now` will wrap around. I.e., this should return
+    /// `core::u32::MAX` on a 32-bit-clock, or `1 << 24` for a 24-bit clock.
+    fn max_tics(&self) -> u32;
```

Comment:
What if it is a 64 bit counter? That is what riscv supports.

Reply:
Does that imply that this whole HIL needs to move over to `u64`?

Reply:
I've not be closely following this PR and related discussions, but thanks for supporting 24-bit counter :-) . 24-bit counter is something we have on our platform.

Reply:
We could make the trait generic in the unit, with a default `u32` or `u64`. So something like:

```rust
pub trait Width {}

impl Width for u32 {};
impl Width for u64 {};

pub trait Time<W: Width = u32> {
    fn now(&self) -> W;
    fn max_tics(&self) -> W;
}
```

Such that `Time` is always sugar for `Time<u32>`, which, I believe, would prevent it from bubbling up to things in the common case.

Reply:
This trait idea seems like the best solution to me, as long as it doesn't make code using the Time trait too hard to write.

The discussion on the call is that using u64 when necessary is important, but using u64 as the default (on 32-bit systems) seems like it would impact performance in a bad way. So the ability to change with traits seems useful.